README.md

```diff
@@ -1,429 +1,139 @@
-      this.schema,
-      this.dataSources.dataSources
-    );
-  }
-
-  public sampleResolve(): Observable<any> {
-    const item = { value: '1', label: 'Art3mis' };
-    return Observable.create((observer: Subject<any>) => {
-      setTimeout(() => {
-        observer.next(item);
-      }, 1000);
-    });
-  }
-
-  public sampleSearch(): Observable<any> {
-    const items: Array<any> = [
-      { value: '0', label: 'Aech' },
-      { value: '5b6e58ea-1359-11df-a1f1-0026b9348838', label: 'Art3mis' },
-      { value: '2', label: 'Daito' },
-      { value: '3', label: 'Parzival' },
-      { value: '4', label: 'Shoto' }
-    ];
-
-    return Observable.create((observer: Subject<any>) => {
-      setTimeout(() => {
-        observer.next(items);
-      }, 1000);
-    });
-  }
-
-  public onSubmit($event) {
-    $event.preventDefault();
-
-    // Set valueProcessingInfo
-    this.form.valueProcessingInfo = {
-      patientUuid: 'patientUuid',
-      visitUuid: 'visitUuid',
-      encounterTypeUuid: 'encounterTypeUuid',
-      formUuid: 'formUuid',
-      encounterUuid: 'encounterUuid',
-      providerUuid: 'providerUuid',
-      utcOffset: '+0300'
-    };
-
-    if (this.form.valid) {
-      this.form.showErrors = false;
-      const payload = this.encAdapter.generateFormPayload(this.form);
-
-      // Alternative is to populate for each as shown below
-      // // generate obs payload
-      // let payload = this.obsValueAdapater.generateFormPayload(this.form);
-
-      // // generate orders payload
-      // let ordersPayload = this.orderAdaptor.generateFormPayload(this.form);
-    } else {
-      this.form.showErrors = true;
-      this.form.markInvalidControls(this.form.rootNode);
-    }
-  }
-
-}
+```sh
+yarn build:lib
 ```
 
-`app.component.html`
+To fire up a dev server, run:
 
-```html
-<div *ngIf="form && form.rootNode">
-  <form [formGroup]="form.rootNode.control">
-    <ofe-form-renderer
-      (onAction)="actionClicked($event)"
-      [node]="form.rootNode"
-    ></ofe-form-renderer>
-  </form>
-</div>
+```sh
+yarn start
 ```
 
-See `src/app/adult-1.4.json` for and `src/app/mock/obs.json` for sample encounter payload
+This should fire up a local server on port `4200`. You can then navigate to `http://localhost:4200/` to see the demo app, which showcases the library's capabilities and provides a playground for testing out new features. The code that powers this demo app is in the `src/app` directory. It renders the form schema defined in `src/app/adult-1.6.json`.
 
-### To publish:
+To link the library to an O3 frontend, run the following commands:
 
-Update the version in both of the `package.json` files
+```sh
+# Build the library
+yarn build:lib
 
-`$ git add -f dist`
+# cd into the dist directory
+cd dist/ngx-formentry
 
-`$ git commit -m 'Bump <Version>'`
+# Create an empty lockfile (yarn 3+ requires this)
+touch yarn.lock
 
-`$ git tag <Version>`
+# Copy the absolute path to the built library
+pwd | pbcopy
```

Comment:
`pbcopy` and `pbpaste` are macOS only. The Linux equivalents are something like:

`pbcopy`
```
xclip -selection clipboard
```

And:

`pbpaste`
```
xclip -out
```

Reply:
So, would that be: `xclip -selection clipboard && xclip -out`?

Reply:
Well, `pwd | xclip -selection clipboard` and then ``yarn add `xclip -out` ``